homeassistant/components/soma/__init__.py

```diff
@@ -0,0 +1,116 @@
+"""Support for Soma Smartshades."""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant import config_entries
+from homeassistant.components.soma import config_flow
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT)
+
+from .const import DOMAIN, HOST, API
+
+DEVICES = 'devices'
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+SOMA_COMPONENTS = ['cover']
+
+
+async def async_setup(hass, config):
+    """Set up the Soma component."""
+    if DOMAIN not in config:
+        return True
+
+    hass.data[DOMAIN] = {}
+
+    config_flow.register_flow_implementation(
+        hass, config[DOMAIN][CONF_HOST],
+        config[DOMAIN][CONF_PORT])
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={'source': config_entries.SOURCE_IMPORT},
```

Comment:
Please add a corresponding import step in the config flow.

Reply:
I think I did that now. No idea how to test this though. What is the use case of this anyway? Is this for when the user has configured this component in configuration.yaml? Ideally I would support that too.

Reply:
Yes!